### PR TITLE
doc: various docstrings

### PR DESCRIPTION
**Pull request by EdAyers**

### CODE REVIEW

File: src/Init/Prelude.lean [hunk 1]
```diff
@@ -961,6 +970,12 @@ theorem usize_size_eq : Or (Eq USize.size 4294967296) (Eq USize.size 18446744073
   | _, Or.inl rfl => Or.inl (by decide)
   | _, Or.inr rfl => Or.inr (by decide)
 
+/-- A USize is an unsigned integer with the size of a word
+for the platform's arctitecture.
```
Comment (gebner) [inline src/Init/Prelude.lean:974]:
```suggestion
for the platform's architecture.
```

File: src/Lean/Meta/MatchUtil.lean [hunk 2]
```diff
@@ -19,6 +19,7 @@ namespace Lean.Meta
   | none => p? (← whnf e)
   | s    => return s
 
+/-- matches `e` with `lhs = rhs : α` and returns `(α, lhs, rhs)`. -/
```
Comment (gebner) [inline src/Lean/Meta/MatchUtil.lean:22]:
```suggestion
/-- Matches `e` with `lhs = (rhs : α)` and returns `(α, lhs, rhs)`. -/
```

File: src/Lean/MetavarContext.lean [hunk 3]
```diff
@@ -420,6 +420,12 @@ def hasAssignableMVar [Monad m] [MonadMCtx m] : Expr → m Bool
 def assignLevelMVar [MonadMCtx m] (mvarId : MVarId) (val : Level) : m Unit :=
   modifyMCtx fun m => { m with lAssignment := m.lAssignment.insert mvarId val, usedAssignment := true }
 
+/--
+  Add `mvarId := u` to the metavariable assignment.
+  This method does not check whether `mvarId` is already assigned, nor it checks whether
+  a cycle is being introduced.
+  This is a low-level API, and it is safer to use `isDefEq (mkLevelMVar mvarId) u`.
```
Comment (gebner) [inline src/Lean/MetavarContext.lean:427]:
```suggestion
Add `mvarId := u` to the metavariable assignment.
This method does not check whether `mvarId` is already assigned, nor it checks whether
a cycle is being introduced, or whether the expression has the right type.
This is a low-level API, and it is safer to use `isDefEq (mkLevelMVar mvarId) u`.
```

File: src/Init/Prelude.lean [hunk 4]
```diff
@@ -858,6 +864,7 @@ instance : Inhabited UInt8 where
   default := UInt8.ofNatCore 0 (by decide)
 
 def UInt16.size : Nat := 65536
+/-- Unsigned, 16-bit integer. -/
```
Comment (gebner) [inline src/Init/Prelude.lean:867]:
```suggestion
/-- Unsigned 16-bit integer. -/
```
And similarly below.

File: src/Lean/Meta/AppBuilder.lean [hunk 5]
```diff
@@ -102,6 +102,7 @@ def mkEqTrans (h₁ h₂ : Expr) : MetaM Expr := do
     | none, _ => throwAppBuilderException ``Eq.trans ("equality proof expected" ++ hasTypeMsg h₁ hType₁)
     | _, none => throwAppBuilderException ``Eq.trans ("equality proof expected" ++ hasTypeMsg h₂ hType₂)
 
+/-- Given `HEq a b`, makes `HEq b a`.  -/
```
Comment (Vtec234) [inline src/Lean/Meta/AppBuilder.lean:105]:
```suggestion
/-- Given `h : HEq a b`, return a proof of `HEq b a`.  -/
```
And similarly for the other ones, to distinguish the proving functions from ones which would manipulate types.

Comment (EdAyers) [reply src/Lean/Meta/AppBuilder.lean]: Should it be `$h : HEq a b`?

Comment (Kha) [reply src/Lean/Meta/AppBuilder.lean]: I don't think we use `$` in this sense in docstrings elsewhere

**Pull request merged**